pkgs/development/libraries/glibc/common.nix

```diff
@@ -64,6 +65,25 @@ stdenv.mkDerivation ({
          "/bin:/usr/bin", which is inappropriate on NixOS machines. This
          patch extends the search path by "/run/current-system/sw/bin". */
       ./fix_path_attribute_in_getconf.patch
+
+      /* Allow running with RHEL 6 -like kernels.  The patch adds an exception
+        for glibc to accept 2.6.32 and to tag the ELFs as 2.6.32-compatible
+        (otherwise the loader would refuse libc).
+        Note that glibc will fully work only on their heavily patched kernels
+        and we lose early mismatch detection on 2.6.32.
```

Comment:
What does it mean to "lose early mismtach detection"?

Reply:
@edolstra: when glibc detect an old kernel, it dies with a readable message (_kernel too old_).  That gets disabled for 2.6.32, so if someone used that version _without_ the RHEL patches it would be hard to debug &ndash; she wouldn't get any readable message and instead, some glibc functions would be failing (`ENOSYS` etc.)

Reply:
I didn't want to make the in-source comment too long, but perhaps I've cut it down too much.